src/audio/ogg_vorbis/castleinternalvorbisdecoder.pas

```diff
@@ -40,9 +40,20 @@   EVorbisFileError = class(EVorbisLoadError);
 function VorbisDecode(Stream: TStream; out DataFormat: TSoundDataFormat;
   out Frequency: LongWord): TMemoryStream;
 
+procedure OpenVorbisFile(var VorbisFile: TOggVorbis_File; Stream: TStream; out DataFormat: TSoundDataFormat; out Frequency: LongWord);
```

Comment:
Can we maybe implement a class like `TOggVorbisStream`, that descends from `TStream` ? It's constructor / `Read` / destructor methods could play the same role as current OpenVorbisFile / ReadVorbisFile / CloseVorbisFile do. Seeking and writing can just raise exceptions.

This would make the code using this even more obvious, as the code would use standard Pascal TStream API.

And then `VorbisDecode` would basically be a function that copies the `TOggVorbisStream` into `TMemoryStream`, which seems sensible, since that's what happens when we "load a file as a whole".

Reply:
This is so internal so... can we move that to another PR? I want now finish my game (ASAP) :)